nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AbstractDatabaseFetchProcessor.java

```diff
@@ -156,10 +157,22 @@
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
             .build();
 
+    public static final PropertyDescriptor SQL_QUERY = new PropertyDescriptor.Builder()
+            .name("db-fetch-sql-query")
+            .displayName("Custom Query")
+            .description("A custom SQL query used to retrieve data. Instead of building a SQL query from "
+                    + "other properties, this query will be used. Query must have no WHERE or ORDER BY statements. "
+                    + "If a WHERE clause is needed use a sub-query or the 'Additional WHERE clause' property.")
```

Comment:
Can we generate a sub-query from this property by simply wrapping it with braces and add the specified `Table name` as the alias name of it? If we do so, we can use Expression Language and incoming FlowFiles more creatively with the single QueryDatabaseTable processor instance. Also it will allow user to define WHERE and ORDER BY statement here.

With that approach, we can keep using the `Table Name` property as it is, and the custom validation is not required. This `Custom Query` property can be truly an optional property.

Current proposed approach use a constant "NiFi_QDBT" table name for state keys, and it can limit the use-case. How do you think?

Reply:
Sounds like a good idea. Users still won't be able to include an ORDER BY, as those are not allowed in sub queries in every database I've worked with.